WcaOnRails/app/views/competitions/_index_competitions_list.html.erb

```diff
@@ -48,19 +48,27 @@
     <% if competitions.empty? %>
       <%= render 'no_competitions_found' %>
     <% else %>
-      <% past_competitions, ongoing_and_upcoming_competitions = competitions.partition(&:is_over?) %>
-      <% recent_competitions, distant_competitions = past_competitions.partition { |c| (Date.today - c.end_date).to_i < Competition::RECENT_DAYS } %>
-      <% unless ongoing_and_upcoming_competitions.empty? %>
-        <div class="col-md-12" id="upcoming-comps">
-          <%= render 'admin_index_table', competitions: ongoing_and_upcoming_competitions, title: "Ongoing and upcoming competitions" %>
-        </div>
+      <% if params[:state] == 'past' %>
+        <% recent_competitions, distant_competitions = competitions.partition { |c| (Date.today - c.end_date).to_i < Competition::RECENT_DAYS } %>
+        <% distant_competitions = distant_competitions.reverse %>
+        <% recent_competitions = recent_competitions.reverse %>
+        <% unless recent_competitions.empty? %>
+          <div class="col-md-12" id="recent-comps">
+            <%= render 'admin_index_table', competitions: recent_competitions, title: "Competitions within the last #{Competition::RECENT_DAYS} days" %>
+          </div>
+        <% end %>
+        <% unless distant_competitions.empty? %>
+          <div class="col-md-12" id="past-comps">
+            <%= render 'admin_index_table', competitions: distant_competitions, title: "Past competitions" %>
+          </div>
+        <% end %>
+      <% else %>
+        <% unless competitions.empty? %>
+          <div class="col-md-12" id="upcoming-comps">
+            <%= render 'admin_index_table', competitions: competitions, title: "Ongoing and upcoming competitions" %>
+          </div>
+        <% end %>
```

Comment:
You can split upcoming and ongoing into two tables if it make sense for you (as you'll be using this).